R/zchunk_LB134.Diet_Rfao.R

```diff
@@ -0,0 +1,384 @@
+      group_by(GCAM_region_ID, year) %>%
+      summarise(value = sum(value)) %>%
+      mutate(GCAM_demand = "meat") %>%
+      bind_rows(L134.ag_Food_Pcal_R_Y) ->
+      L134.Food_Pcal_R_Dmnd_Y
+
+    # Divide by population to calculate the historical per-capita food demands, in kcal per person per day
+    # Original lines 51-56
+    L134.Food_Pcal_R_Dmnd_Y %>%
+      left_join_error_no_match(L101.Pop_thous_R_Yh, by = c("GCAM_region_ID", "year")) %>%
+      mutate(value = value.x * CONV_DAYS_YEAR * (1 / CONV_MCAL_PCAL ) / value.y) %>%
+      select(-value.x, -value.y) ->
+      L134.pcFood_kcald_R_Dmnd_Y
+
+    # In preparation for future extrapolation, get the last historical numbers and aggregate by ISO
+    # Original lines 58-69
+    L100.FAO_ag_Food_t %>%
+      filter(year == max(HISTORICAL_YEARS)) %>%
+      group_by(iso) %>%
+      summarise(value = sum(value)) %>%
+      mutate(GCAM_demand = "crops") ->
+      L134.FAO_ag_Food_t
+
+    L100.FAO_an_Food_t %>%
+      filter(year == max(HISTORICAL_YEARS)) %>%
+      group_by(iso) %>%
+      summarise(value = sum(value)) %>%
+      mutate(GCAM_demand = "meat") %>%
+      bind_rows(L134.FAO_ag_Food_t) %>%
+      left_join_keep_first_only(select(AGLU_ctry, FAO2050_reg, iso), by = "iso") ->
+      L134.Food_t_ctry_Dmnd_Y
+
+    # Calculate the future demand ratios by FAO2050 region for each demand type
+    # Drop unnecessary composite regions from FAO diet table
+    # Original lines 71-85
+    FAO2050_Diet %>%
+      filter(FAO2050_reg %in% L134.Food_t_ctry_Dmnd_Y$FAO2050_reg) %>%
+      # Calculate FAO2050 diet for specified diet years
+      # Use linear interpolation to convert FAO2050 model time periods to GCAM "diet years"
+      group_by(FAO2050_reg, FAO2050_item) %>%
+      # Extend years in dataset to include DIET_YEARS
+      complete(year = unique(c(.$year, aglu.DIET_YEARS))) %>%
+      mutate(value = approx_fun(year, value)) %>%
+      # Add caloric content and demand category
+      left_join(select(FAO2050_items_cal, FAO2050_item, GCAM_demand, kcalkg, conv_d), by = "FAO2050_item") ->
+      L134.Diet_Rfao_Cfao_Yfao
+
+    # Build new table with only diet years subsetted
+    # Multiply through by caloric contents to get all measures in kcal/pers/d, by FAO region and food categories
+    # Original lines 87-92
+    L134.Diet_Rfao_Cfao_Yfao %>%
+      filter(year %in% aglu.DIET_YEARS) %>%
+      mutate(value = value * kcalkg / conv_d) %>%
+      select(FAO2050_reg, FAO2050_item, GCAM_demand, year, value) ->
+      L134.Diet_kcald_Rfao_Cfao_Y
+
+    # Aggregate by GCAM demand and FAO region, and compute future diet ratios by FAO2050 region
+    # Original lines 94-104
+    L134.Diet_kcald_Rfao_Cfao_Y %>%
+      filter(year %in% aglu.DIET_YEARS, !is.na(GCAM_demand)) %>%
+      group_by(FAO2050_reg, GCAM_demand, year) %>%
+      summarise(value = sum(value)) %>%
+      ungroup %>%
+      mutate(GCAM_demand = if_else(GCAM_demand == "total", "crops", GCAM_demand)) ->
```

Comment:
Good question, thanks. @pkyle can you explain reasoning / logic? Why are we replacing all "total" with "crops" here?